### PR TITLE
[Bindless][Exp] Create mapping of host and device handle

### PR DESCRIPTION
**Pull request by DBDuncan**
Creates a mapping between the host and device handle in the L0 backend to be able to retrieve the host handle when destroying the image.

### CODE REVIEW

**Comment by DBDuncan** [comment]
Corresponding UR DPC++: https://github.com/intel/llvm/pull/14306

File: source/adapters/level_zero/image.cpp [hunk 1]
```diff
@@ -675,8 +677,15 @@ urBindlessImagesUnsampledImageHandleDestroyExp(ur_context_handle_t hContext,
                                                ur_device_handle_t hDevice,
                                                ur_exp_image_handle_t hImage) {
   std::ignore = hContext;
-  std::ignore = hDevice;
-  std::ignore = hImage;
+
```
Comment (nrspruit) [inline source/adapters/level_zero/image.cpp:680]: Can you add the UR_ASSERT checks for the handles in this api?

ie:
UR_ASSERT(hContext && hDevice && hImage, UR_RESULT_ERROR_INVALID_NULL_HANDLE);

Then you can also remove the std::ignore = hContext

**Review by cppchedy — APPROVED** [review]

File: source/adapters/level_zero/image.cpp (hunk 1 above)
Comment (DBDuncan) [reply source/adapters/level_zero/image.cpp:680]: Done.

**Comment by DBDuncan** [comment]
Friendly ping @nrspruit . I wonder why the `unified-runtime-level-zero-write ` group was not automaticity added to this PR but was to another PR I have. A bit odd.

**Comment by DBDuncan** [comment]
Friendly ping @oneapi-src/unified-runtime-level-zero-write

**Review by pbalcer — APPROVED** [review]
lgtm, but the CI was failing. I restarted it, but please take a look if it continues to fail

**Comment by DBDuncan** [comment]
@nrspruit @oneapi-src/unified-runtime-level-zero-write Reviews please

**Comment by kbenzie** [comment]
`source/adapters/level_zero/device.hpp` has a conflict

**Pull request merged**